src/coreclr/jit/lowerxarch.cpp

```diff
@@ -1631,6 +1779,27 @@ GenTree* Lowering::LowerHWIntrinsic(GenTreeHWIntrinsic* node)
             LowerFusedMultiplyAdd(node);
             break;
 
+        case NI_AVX_BroadcastScalarToVector256:
+        case NI_AVX_BroadcastScalarToVector128:
+        {
+            // there can be 2 cases hitting AVX_BroadcastScalarToVector*
+            // 1. pass the address as LCL_ADDR to AVX.BroadcastScalarToVector256() API
+            // 2. pass the address as LCL_VAR  to AVX.BroadcastScalarToVector256() API
```

Comment:
I think this is missing some cases such as `GT_IND` which can probably occur if the input is a `T*` parameter passed into the caller.

Reply:
Do you mean code like this?
```C#
[MethodImplAttribute(MethodImplOptions.NoInlining)]
public unsafe static Vector256<float> Foo(float* fAddr)
{
	Vector256<float> Vec = Avx.BroadcastScalarToVector256(fAddr);
	Vector256<float> VecCns = Vector256.Create(1.0f,2.0f,3.0f,4.0f,5.0f,6.0f,7.0f,8.0f);
	return Vector256.Add(Vec, VecCns);
}
```
And this method be called like:
```C#
unsafe{
	float f = 5.0f;
	Foo(&f);
}

```
This `fAddr` will be imported as a local variable. If there are some other cases, can you please give an example?

Reply:
Ah, thanks, Will cover this case.

Reply:
Looks like the `GT_IND` touches the API in such styles: [BroadcastScalarToVector256(Vector128<Single>)](https://learn.microsoft.com/en-us/dotnet/api/system.runtime.intrinsics.x86.avx2.broadcastscalartovector256?view=net-7.0#system-runtime-intrinsics-x86-avx2-broadcastscalartovector256(system-runtime-intrinsics-vector128((system-single)))), so the related changes are in `NI_AVX2_BroadcastScalarToVector*`, and the AVX version won't be updated. Is that expected?

Reply:
That'd be my expectation. `IND` represents an indirection, that is effectively a dereferenced pointer. Since the `AVX` versions can only take an address, we shouldn't be able to see an indirection at all.

For the ones that take a value, we should be able to see `IND`, `LCL_FLD`, or `LCL_VAR`. -- More concretely its anything for which `LinearScan::isContainableMemoryOp` returns `true`. That is currently:
* `GenTree::isMemoryOp`
  * `GenTree::isIndir()` -- `GT_IND`
  * `GenTree::isLclField()` -- `GT_LCL_FLD`
* `GenTree::IsLocal()` where `LinearScan::enregisterLocalVars == false` or `lvaGetDesc(node)->lvDoNotRegister == true`
  * This is functionally just `GT_LCL_VAR` given the `isMemoryOp` check